### PR TITLE
arch/arm/toolchain: migrate the toolchain define to arch/arm/Kconfig

### PR DESCRIPTION
**Pull request by anchao**
## Summary

arch/arm/toolchain: migrate the toolchain define to arch/arm/Kconfig

migrate the toolchain define to arch/arm/Kconfig to simplify new toolchain registration

Signed-off-by: chao an <anchao@xiaomi.com>

## Impact

N/A

## Testing

ci-check

### CODE REVIEW

File: arch/Kconfig [hunk 1]
```diff
@@ -230,57 +230,7 @@ endmenu # Custom Chip Configuration
...
-		This option enables Clang's ThinLTO, which allows for parallel
-		optimization and faster incremental compiles compared to the
-		CONFIG_LTO_FULL option. More information can be found
-		from Clang's documentation:
-
-		https://clang.llvm.org/docs/ThinLTO.html
-
-		If unsure, say Y.
-endchoice
+source "arch/Kconfig.toolchain"
```
Comment (xiaoxiang781216) [inline arch/Kconfig:233]: why split to Kconfig.toolchain?

File: arch/Kconfig.toolchain [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	---help---
+		This option should work for any GNU toolchain.
+
+config TOOLCHAIN_CLANG
+	bool "Generic Clang toolchain"
+	select ARCH_TOOLCHAIN_CLANG
+
+endchoice
+
+config TOOLCHAIN_BUILDROOT_OABI
```
Comment (xiaoxiang781216) [inline arch/Kconfig.toolchain:51]: merge to choice?

File: arch/Kconfig.toolchain [hunk 3]
```diff
@@ -0,0 +1,94 @@
+#
+# For a description of the syntax of this configuration file,
+# see the file kconfig-language.txt in the NuttX tools repository.
+#
+
+config ARCH_TOOLCHAIN_IAR
```
Comment (xiaoxiang781216) [inline arch/Kconfig.toolchain:6]: remove check TOOLCHAIN_IAR instead?

File: arch/Kconfig.toolchain [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+	bool
+	default n
+
+config ARCH_TOOLCHAIN_GNU
```
Comment (xiaoxiang781216) [inline arch/Kconfig.toolchain:10]: TOOLCHAIN_IS_GNU

File: arch/Kconfig.toolchain [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+	bool
+	default n
+
+config ARCH_TOOLCHAIN_CLANG
```
Comment (xiaoxiang781216) [inline arch/Kconfig.toolchain:14]: TOOLCHAIN_IS_CLANG

File: arch/Kconfig (hunk 1 above)
Comment (anchao) [reply arch/Kconfig:233]: Just separates the toolchain related configuration into Kconfig.toolchain, new toolchain support can be added to this file

File: arch/Kconfig.toolchain (hunk 2 above)
Comment (anchao) [reply arch/Kconfig.toolchain:51]: let us keep the original configuration for this PR, BUILDROOT_OABI needs some additional changes

File: arch/Kconfig.toolchain (hunk 5 above)
Comment (anchao) [reply arch/Kconfig.toolchain:14]: Done

File: arch/Kconfig.toolchain (hunk 4 above)
Comment (anchao) [reply arch/Kconfig.toolchain:10]: Done

File: arch/Kconfig.toolchain (hunk 3 above)
Comment (anchao) [reply arch/Kconfig.toolchain:6]: Done

File: arch/Kconfig.toolchain (hunk 2 above)
Comment (xiaoxiang781216) [reply arch/Kconfig.toolchain:46]: but this change is already huge, you even rename ARM_OABI_TOOLCHAIN to TOOLCHAIN_BUILDROOT_OABI

Comment (anchao) [reply arch/Kconfig.toolchain:46]: OK

**Comment by xiaoxiang781216** [comment]
let's track by https://github.com/apache/incubator-nuttx/pull/7100?

**Pull request closed without merging**